### ISSUE TITLE
Exception on deserialization of an official xml sample

### ISSUE THREAD

**Issue by raffaeler** [FatturaElettronica/FatturaElettronica.Core#4]
The sample is basically the same of the FatturaElettronica.NET home page:
https://github.com/FatturaElettronica/FatturaElettronica.NET
The only difference is that is running inside an ASP.NET Core 2.1 application.

I posted here the issue since the call stack points to the `FatturaElettronica.Common.BaseClassSerializable` class.

### Sample
```
private Task<bool> ProcessFile(FileInfo fileInfo)
{
    // Usare il factory method CreateInstance() per ottenere una istanza di Fattura.
    var fattura = Fattura.CreateInstance(Instance.PubblicaAmministrazione);

    // Lettura da file XML
    using (var r = XmlReader.Create(fileInfo.FullName, new XmlReaderSettings { IgnoreWhitespace = true, IgnoreComments = true }))
    {
        fattura.ReadXml(r);
    }

    // Ogni file di fattura contiene un array di elementi FatturaElettronicaBody.
    _logger.LogInformation($"Numero di documenti: {fattura.Body.Count}.");
    _logger.LogInformation("Documenti inclusi nel file FatturaPA:");
    foreach (var doc in fattura.Body)
    {
        _logger.LogInformation($"Numero: {doc.DatiGenerali.DatiGeneraliDocumento.Numero}");
        _logger.LogInformation($"Data: {doc.DatiGenerali.DatiGeneraliDocumento.Data.ToShortDateString()}");
        _logger.LogInformation($"Totale documento: {doc.DatiGenerali.DatiGeneraliDocumento.ImportoTotaleDocumento}");
        _logger.LogInformation("");
    }

    // Convalida del documento.
    var validator = new FatturaValidator();
    var result = validator.Validate(fattura);
    _logger.LogInformation(result.IsValid ? "Validated!" : "Xml is not valid");

    // Introspezione errori di convalida.
    foreach (var error in result.Errors)
    {
        _logger.LogError(error.PropertyName);
        _logger.LogError(error.ErrorMessage);
        // Nei casi di errore 2xx e 4xx ErrorCode conterrà il codice errore (es: "00423").
        _logger.LogError(error.ErrorCode);
    }

    if (!result.IsValid)
        return Task.FromResult(false);

    return Task.FromResult(true);
}
```

The tested file is the one provided as a sample here:
http://www.fatturapa.gov.it/export/fatturazione/it/a-3.htm

### Call stack
```
System.Xml.XmlException: Content cannot be converted to the type System.DateTime. Line 131, position 44. ---> System.FormatException: The string '2017-02-30' is not a valid AllXsd value.
   at System.Xml.Schema.XsdDateTime..ctor(String text, XsdDateTimeFlags kinds)
   at System.Xml.Schema.XmlUntypedStringConverter.FromString(String value, Type destinationType, IXmlNamespaceResolver nsResolver)
   at System.Xml.XmlReader.ReadContentAs(Type returnType, IXmlNamespaceResolver namespaceResolver)
   --- End of inner exception stack trace ---
   at System.Xml.XmlReader.ReadContentAs(Type returnType, IXmlNamespaceResolver namespaceResolver)
   at System.Xml.XmlReader.ReadElementContentAs(Type returnType, IXmlNamespaceResolver namespaceResolver)
   at FatturaElettronica.Common.BaseClassSerializable.ReadXml(XmlReader r)
   at FatturaElettronica.Common.BaseClassSerializable.ReadXmlList(Object propertyValue, Type propertyType, String elementName, XmlReader r)
   at FatturaElettronica.Common.BaseClassSerializable.ReadXml(XmlReader r)
   at FatturaElettronica.Common.BaseClassSerializable.ReadXmlList(Object propertyValue, Type propertyType, String elementName, XmlReader r)
   at FatturaElettronica.Common.BaseClassSerializable.ReadXml(XmlReader r)
   at FatturaElettronica.Common.BaseClassSerializable.ReadXmlList(Object propertyValue, Type propertyType, String elementName, XmlReader r)
   at FatturaElettronica.Common.BaseClassSerializable.ReadXml(XmlReader r)
   at FatturaElettronica.Fattura.ReadXml(XmlReader r)
   at ProcessFile(FileInfo fileInfo) in ...
```

### More Info.
The user profile is configured with en-us settings. My guess is that the FatturaElettronica.Core assumes the localization happens with it-it settings (which rarely happens in case of server deploy such as Windows Service or Linux container).

### Suggestion.
Please consider using the Linq to Xml (XElement  / XDocument) classes instead of System.Xml set of classes.

**Comment by nicolaiarocci** [comment]
hey Raf 😄

**Comment by raffaeler** [comment]
I may have found the problem.
You can add the `XmlElement` attribute on all the DateTime properties and specify the "Qualified" schema form:
```
  [System.Xml.Serialization.XmlElementAttribute(Form=System.Xml.Schema.XmlSchemaForm.Qualified, DataType="date")]
    public System.DateTime DataScadenzaPagamento { ... }
```

I didn't clone your repo, but I generated the classes with xsd.exe and got a repro with the same problem. Then I fixed the issue by changing Unqualified to Qualified.

HTH

**Comment by nicolaiarocci** [comment]
Got it. The 'official' xml example (the one you also mailed me) is bogus: the date on which it crashes is impossible (30 February, 2017). That file is actually the same we have in the test suite, and it was running just fine... but then I remembered that 4 years ago I had to fix the bogus date manually, when I found out.

So long story short, if you go to the xml and fix line 131 to '2017-02-28', then it will se(de)rialize just fine.
